napari/_qt/dialogs/qt_plugin_dialog.py

```diff
@@ -164,69 +247,167 @@ def setup_ui(self, enabled=True):
+
+        info_layout = QGridLayout()
+        info_layout.setContentsMargins(0, 0, 0, 0)
+        info_layout.setVerticalSpacing(2)
+        info_layout.setHorizontalSpacing(4)
+        info_layout.addWidget(self.source_choice_text, 0, 0)
+        info_layout.addWidget(self.source_choice_dropdown, 1, 0)
+        info_layout.addWidget(self.version_choice_text, 0, 1)
+        info_layout.addWidget(self.version_choice_dropdown, 1, 1)
+        self.info_choice_wdg.setLayout(info_layout)
+        self.info_choice_wdg.setLayoutDirection(Qt.LeftToRight)
+        self.info_choice_wdg.setObjectName("install_choice_widget")
+        self.row2.addWidget(
+            self.info_choice_wdg, alignment=Qt.AlignmentFlag.AlignTop
+        )
+        self.info_choice_wdg.hide()
+
+        self.cancel_btn = QPushButton("Cancel", self)
+        self.cancel_btn.setSizePolicy(QSizePolicy.Fixed, QSizePolicy.Fixed)
+        self.cancel_btn.setObjectName("remove_button")
+        self.row2.addWidget(
+            self.cancel_btn, alignment=Qt.AlignmentFlag.AlignTop
+        )
+
+        self.action_button = QPushButton(self)
+        sizePolicy = QSizePolicy(QSizePolicy.Fixed, QSizePolicy.Fixed)
         sizePolicy.setHorizontalStretch(0)
         sizePolicy.setVerticalStretch(0)
         sizePolicy.setHeightForWidth(
-            self.package_author.sizePolicy().hasHeightForWidth()
+            self.action_button.sizePolicy().hasHeightForWidth()
         )
-        self.package_author.setSizePolicy(sizePolicy)
-        self.package_author.setObjectName("small_text")
-        self.row2.addWidget(self.package_author)
+
+        self.action_button.setSizePolicy(sizePolicy)
+        self.row2.addWidget(
+            self.action_button, alignment=Qt.AlignmentFlag.AlignTop
+        )
+
         self.v_lay.addLayout(self.row2)
 
+        self.info_widget = QWidget(self)
+        self.info_widget.setLayoutDirection(Qt.LeftToRight)
+        self.info_widget.setObjectName("info_widget")
+        info_layout = QGridLayout()
+        info_layout.setContentsMargins(0, 0, 0, 0)
+        info_layout.setVerticalSpacing(0)
+        self.version_text = QLabel('Version:')
+        self.package_name = QLabel()
+        self.source_text = QLabel('Source:')
+        self.source = QLabel('PyPI')
+
+        info_layout.addWidget(self.source_text, 0, 0)
+        info_layout.addWidget(self.source, 1, 0)
+        info_layout.addWidget(self.version_text, 0, 1)
+        info_layout.addWidget(self.package_name, 1, 1)
+
+        self.install_info_button.layout().setContentsMargins(0, 0, 0, 0)
+        self.info_widget.setLayout(info_layout)
+
+    def _populate_version_dropdown(self, source):
+        """Display the versions available after selecting a source: pypi or conda."""
+
+        versions = self._versions[source]
+        self.version_choice_dropdown.clear()
+        if len(versions) > 0:
+            for version in versions:
+                self.version_choice_dropdown.addItem(version)
```

Comment:
```suggestion
        for version in versions:
            self.version_choice_dropdown.addItem(version)
```
If there is no entry in versions, then for loop will not execute. The `if` is obsolete.

Reply:
bump this remark